lib/workers/repository/process/lookup/index.js

```diff
@@ -37,6 +37,21 @@ async function lookupUpdates(config) {
     res.updates.push(result);
     return res;
   }
+  const { deprecationMessage, deprecationSource } = dependency;
+  // istanbul ignore if
+  if (deprecationMessage && config.raiseDeprecationWarnings) {
+    logger.info(
+      {
+        deprecationSource,
+        depName,
+        deprecationMessage,
+      },
+      'dependency is deprecated'
+    );
+    const issueTitle = `Dependency deprecation warning: ${depName} (${deprecationSource})`;
+    const issueBody = `This project is using package \`${depName}\` as a dependency, however it is marked as deprecated on ${deprecationSource}. The deprecation message is: \`${deprecationMessage}\`\n\nPlease take the actions necessary to rename or substitute this deprecated package and commit to your base branch. If you wish to ignore this deprecation warning and continue using \`${depName}\` as-is, please add it to your [ignoreDeps](https://renovatebot.com/docs/configuration-options/#ignoredeps) array in Renovate config before closing this issue, otherwise another issue will be recreated the next time Renovate runs.`;
```

Comment:
> `marked as deprecated on ${deprecationSource}`

This will make sense for general consumers of the Renovate App, but may not make sense for Enterprise users (since some use Artifactory/Nexus/etc.)

Reply:
So just leave source out? I think we need to differentiate between dependency *types* (e.g. npm, pypi, packagist, docker, etc) in case we support others in future.